dds_cli/data_putter.py

```diff
@@ -169,6 +170,75 @@ def put(
                         for x in [y.id for y in progress.tasks if y.fields.get("step") != "put"]
                     ]
 
+        # Make a single database update for files that have failed
+        # Json file for failed files should only be created if there has been an error
+        if putter.failed_delivery_log.is_file():
+            LOG.warning(
+                "Some uploaded files could not be added to the database %s.\n",
+                str(putter.failed_delivery_log),
+            )
+            LOG.warning("Attempting to add them to the database.")
+
+            with putter.failed_delivery_log.open(mode="r", encoding="utf-8") as json_f:
+                failed = json.load(json_f)
```

Comment:
this would throw an exception if there is some error in the json or for some reason the failed files are empty. Could be good to encapsulate it in a try catch if it is not already handled in upper functions